From 95938157db58cb95dd0558f2881131f28b7c41c8 Mon Sep 17 00:00:00 2001
From: liutianyu1250 <liutianyu1250@phytium.com.cn>
Date: Fri, 7 Feb 2025 17:39:27 +0800
Subject: [PATCH 78/81] cpufreq: phytium: Fix Phytium P-state driver build fail
 in v6.6.63

After v6.6.58, cpufreq/cppc already export cppc_perf_to_khz and
cppc_khz_to_perf, no need add in phytium pstate driver.

Signed-off-by: liutianyu1250 <liutianyu1250@phytium.com.cn>
---
 drivers/cpufreq/phytium_pstate.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/cpufreq/phytium_pstate.c b/drivers/cpufreq/phytium_pstate.c
index f937ac56a..54a6a9731 100644
--- a/drivers/cpufreq/phytium_pstate.c
+++ b/drivers/cpufreq/phytium_pstate.c
@@ -232,6 +232,8 @@ static int phyt_pstate_set_energy_pref_index(struct phyt_cpudata *cpudata,
 	return ret;
 }
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,6,58)// in v6.6.58, cppc_acpi already export these API below
 unsigned int cppc_perf_to_khz(struct cppc_perf_caps *caps, unsigned int perf)
 {
 	s64 retval, offset = 0;
@@ -271,6 +273,7 @@ unsigned int cppc_khz_to_perf(struct cppc_perf_caps *caps, unsigned int freq)
 		return retval;
 	return 0;
 }
+#endif
 
 static int phyt_pstate_init_perf(struct phyt_cpudata *cpudata)
 {
-- 
2.48.1

